packages/components/src/globals/scss/_tooltip.scss

```diff
@@ -71,32 +71,28 @@
   }
 }
 
-// Tooltip Icon caret - top position
+// Tooltip Icon placement
+/// @param {String} $position ['bottom'] - The position from: `top`, `bottom`
 /// @access public
 /// @group tooltip
-@mixin tooltip--icon--top {
+@mixin tooltip--icon-placement($position: 'bottom') {
   &::before {
-    top: 1px;
-    transform: translate(-50%, calc(-100% - 9px)) rotate(180deg);
-  }
-
-  &::after {
-    top: 0;
-    transform: translate(-50%, calc(-100% - 12px));
-  }
-}
-
-// Tooltip Icon caret - bottom position
-/// @access public
-/// @group tooltip
-@mixin tooltip--icon--bottom {
-  &::before {
-    bottom: 0;
-    transform: translate(-50%, 10px);
+    @if ($position == 'top') {
+      top: 1px;
+      transform: translate(-50%, calc(-100% - 9px)) rotate(180deg);
```

Comment:
Could you leave a comment explaining where the pixel values come from for positioning/translating? Will help a ton when looking at this in the future 👍

Reply:
@emyarod Do you remember this topic? Thanks!

Reply:
I just tweaked the existing style rules, and it looks like as it is now, those rules will need to change based on the parent element's dimensions. I will refactor this to make the positioning rules more universal

Reply:
🙇 I should have refactored the tooltip positioning logic when I saw it, but here is a PR for refactoring top/bottom positioning https://github.com/carbon-design-system/carbon/pull/2749

similar translate rules can be used for left/right positioning, except the translateX and translateY values will be swapped, so something like this:

```css
// Tooltip Icon caret - left position
/// @access public
/// @group tooltip
@mixin tooltip--icon--left {
  &::before,
  &::after {
    top: 50%;
    left: 0;
  }

  &::before {
    transform: translate(calc(-100% - #{$tooltip-caret-spacing}), -50%);
  }

  &::after {
    transform: translate(
      calc(-100% - #{$tooltip-caret-spacing} - 5px)
      -50%,
    ); // 5px caret height
  }
}

// Tooltip Icon caret - right position
/// @access public
/// @group tooltip
@mixin tooltip--icon--right {
  &::before,
  &::after {
    top: 50%;
    right: 0;
  }

  &::before {
    transform: translate(calc(100% + #{$tooltip-caret-spacing}), -50%);
  }

  &::after {
    transform: translate(
      calc(100% + #{$tooltip-caret-spacing} + 5px)
      -50%,
    ); // 5px caret height
  }
}
```